packages/venia-ui/lib/drivers/adapter.js

```diff
@@ -38,6 +39,65 @@ persistCache({
     storage: window.localStorage
 });
 
+const initialData = {
+    __typename: 'CartData',
+    gift_options: {
+        __typename: 'GiftOptions',
+        include_gift_receipt: false,
+        include_printed_card: false,
+        gift_message: ''
+    }
+};
+
+const GET_GIFT_OPTIONS_QUERY = gql`
+    query getGiftOptions {
+        gift_options @client {
+            __typename
+            include_gift_receipt
+            include_printed_card
+            gift_message
+        }
+    }
+`;
+
+const resolvers = {
+    Query: {
+        gift_options: (_, __, { cache }) => {
+            const {
+                gift_options: {
+                    __typename,
+                    include_gift_receipt,
+                    include_printed_card,
+                    gift_message
+                }
+            } = cache.readQuery({
+                query: GET_GIFT_OPTIONS_QUERY
+            });
+            return {
+                __typename,
+                include_gift_receipt,
+                include_printed_card,
+                gift_message
+            };
+        }
+    },
+    Mutation: {
+        set_gift_options: (_, variables, { cache }) => {
+            const prevData = cache.readQuery({ query: GET_GIFT_OPTIONS_QUERY });
+            cache.writeData({
+                data: {
+                    gift_options: {
```

Comment:
You'll want to read and write this data to the current cart rather than the root.